api/v1beta1/infraenv_types.go

```diff
@@ -88,6 +88,12 @@ type InfraEnvSpec struct {
 	// +kubebuilder:default=x86_64
 	// +optional
 	CpuArchitecture string `json:"cpuArchitecture,omitempty"`
+
+	// IPXEScriptType the script type that should be served (HostRedirect/Boot)
+	// +kubebuilder:default=Boot
+	// +kubebuilder:validation:Pattern=Boot|HostRedirect
```

Comment:
I think the Enum validation is a better option since it enables UIs to present the acceptable values and let the user pick one.

Reply:
Done